core/base/abstractTriangulation/AbstractTriangulation.h

```diff
@@ -2760,6 +2808,38 @@ namespace ttk {
       return &(this->vertexGidToLid_);
     }
 
+    virtual inline const std::unordered_map<SimplexId, SimplexId> &
+      getEdgeGlobalIdMap() const {
+#ifndef TTK_ENABLE_KAMIKAZE
+      if(this->getDimensionality() != 1 && this->getDimensionality() != 2
+         && this->getDimensionality() != 3) {
+        this->printErr("Only 1D, 2D and 3D datasets are supported");
+      }
+      if(!this->hasPreconditionedDistributedEdges_) {
+        this->printErr("VertexGlobalMap query without pre-process!");
+        this->printErr(
+          "Please call preconditionDistributedEdges() in a pre-process.");
+      }
+#endif // TTK_ENABLE_KAMIKAZE
+      return this->edgeGidToLid_;
+    }
+
+    virtual inline const std::unordered_map<SimplexId, SimplexId> &
+      getTriangleGlobalIdMap() const {
+#ifndef TTK_ENABLE_KAMIKAZE
+      if(this->getDimensionality() != 1 && this->getDimensionality() != 2
+         && this->getDimensionality() != 3) {
+        this->printErr("Only 1D, 2D and 3D datasets are supported");
+      }
+      if(!this->hasPreconditionedDistributedEdges_) {
+        this->printErr("VertexGlobalMap query without pre-process!");
+        this->printErr(
+          "Please call preconditionDistributedTriangles() in a pre-process.");
+      }
+#endif // TTK_ENABLE_KAMIKAZE
+      return this->triangleGidToLid_;
+    }
+
```

Comment:
These methods provide an external access to the corresponding class members, which might be overkill if only used inside `ExplicitTriangulation`.

Reply:
I will move them to `ExplicitTriangulation`.

Reply:
If you don't need these methods outside the `ExplicitTriangulation`, I think there is no need to expose the class members this way. I would suggest to remove these methods.